src/features/download_sentinel2_mosaics.py

```diff
@@ -0,0 +1,413 @@
+            cloud_mask = cloud_detector.get_cloud_masks(img_batch)[0]
+            combined_mask = cloud_mask.astype(bool)
+        except Exception as e:
+            logging.error(f"Cloud mask failed for {prefix}: {e}")
+            combined_mask = np.zeros(img.shape[1:], dtype=bool)
+
+        img_with_indices[:, combined_mask] = NO_DATA
+
+        meta_list.append({
+            "date_range": [s, e],
+            "cloud_fraction": float(combined_mask.mean()),
+            "mean_ndvi": float(ndvi[ndvi != NO_DATA].mean()) if np.any(ndvi != NO_DATA) else NO_DATA,
+            "mean_evi": float(evi[evi != NO_DATA].mean()) if np.any(evi != NO_DATA) else NO_DATA,
+            "mean_ndwi": float(ndwi[ndwi != NO_DATA].mean()) if np.any(ndwi != NO_DATA) else NO_DATA 
+        })
+        stack_list.append(img_with_indices)
+
+    return stack_list, meta_list
+
+def retrieve_images():
+    '''
+    For each labeled image, downloads its corresponding time series (36 images, with 18 images before the label date
+    and 18 images after the label date) into ~/data/features/
+    
+    To do this, we attempt to retrieve images from the Google Cloud Bucket. If the image is not there, we request
+    the image from Google Earth Engine, processes each image into the correct format (through calculation of 
+    vegetation bands, use of cloud mask), add it to Google Cloud Bucket, and retrieve it from there.
+    '''
+    logging.basicConfig(level=logging.INFO)
+    os.makedirs(TMP_DIR, exist_ok=True)
+    data = pd.read_csv(LABEL_CSV)
+    initialize_earthengine()
+    for idx, row in data.iterrows():
+        lat, lon = row['y'], row['x']
+        uid = row['unique_id']
+        date = datetime(int(row['year']), int(row['month']), int(row['day']))
+        site_id = f"site_{lat:.2f}_{lon:.2f}_{date.year}_{uid}"
+        stack_list, meta_list = retrieve_time_series_stack(site_id, lat, lon, date)
+        stack_arr = np.stack(stack_list)
+        T, B, H, W = stack_arr.shape
+        reshaped = stack_arr.transpose(1, 0, 2, 3).reshape(T*B, H, W)
+        out_tif = os.path.join(DOWNLOAD_DIR, f"{site_id}.tif")
+        out_json = os.path.join(DOWNLOAD_DIR, f"{site_id}.json")
+
+        with rasterio.open(out_tif, 'w', driver='GTiff', height=H, width=W, count=T*B,
+                        dtype='int16', crs='EPSG:32633',
+                        transform=from_origin(lon - 0.0005, lat + 0.0005, 0.0001, 0.0001),
+                        nodata=NO_DATA) as dst:
+            dst.write(reshaped.astype('int16'))
+
+        with open(out_json, 'w') as f:
+            json.dump({
+                "site_id": site_id,
+                "lat": lat, "lon": lon,
+                "year": date.year, 
+                "unique_id": uid,
+                "bands": FINAL_BANDS,
+                "shape": list(stack_arr.shape),
+                "windows": meta_list
+            }, f, indent=2)
+
+        logging.info(f"[DONE] Saved stack to {out_tif}, shape={stack_arr.shape}")
```

Comment:
Just for safety can we have this throw an error if the size of the tif is not what we end up needing? (13*37, 100, 100 or whatever it's supposed to be)